src/boot/AppDataFetcher.js

```diff
@@ -14,26 +14,23 @@ type Props = $ReadOnly<{|
   children: React$Node,
 |}>;
 
-class AppDataFetcher extends PureComponent<Props> {
-  componentDidUpdate() {
-    const { dispatch, needsInitialFetch } = this.props;
+function AppDataFetcher(props: Props) {
+  const { needsInitialFetch, dispatch } = props;
 
+  React.useEffect(() => {
```

Comment:
One thing I'm not so sure about is how `shouldComponentUpdate` and `PureComponent`s get translated to Hooks.

One thing we could use, in the direction of getting `PureComponent`'s optimizations, is [`React.memo`](https://reactjs.org/docs/react-api.html#reactmemo). While `PureComponent` does a shallow comparison of both props and state, `React.memo` only shallowly compares props. Not using `React.memo`, I believe, is comparable to using `Component` instead of `PureComponent`.

Can you think of any advantage we've been currently getting by using `PureComponent` for `AppDataFetcher`? I'm a little stuck on that question when I look at our pattern of using `children` as a render prop. My first thought is that we're probably not getting much of an advantage, if any. But you're right, it would be educational to find out.

Reply:
I'm not sure how the following translates to React Native, but in React on the Web, [`useEffect`](https://reactjs.org/docs/hooks-reference.html#useeffect) is unlike `componentDidMount` in that it fires after the browser has painted. To avoid this change, we could use [`useLayoutEffect`](https://reactjs.org/docs/hooks-reference.html#uselayouteffect), but the docs specifically recommend trying `useEffect` first:

> If you’re migrating code from a class component, note `useLayoutEffect` fires in the same phase as `componentDidMount` and `componentDidUpdate`. However, **we recommend starting with useEffect first** and only trying `useLayoutEffect` if that causes a problem.

Reply:
Looking through a TL;DR on a [blog post by Dan Abramov on `useEffect`](https://overreacted.io/a-complete-guide-to-useeffect/#tldr), it seems smart to generally assume there will be functional changes when you convert a class component to one that uses Hooks.

Reply:
I replied [in chat](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/converting.20to.20Hooks/near/1092349), to make more room and because this seems like a discussion with reusable interest.